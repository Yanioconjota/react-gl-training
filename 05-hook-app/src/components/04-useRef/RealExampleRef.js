import React, { useState } from "react";
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  return (
    <>
      <h1>Real Example Ref</h1>
      <hr/>

      {show && <MultipleCustomHooks/>}

      <button
        className="btn btn-primary mt-4"
        onClick={()=>{setShow(!show)}}>Show/Hide</button>

    </>
  );
};

export default RealExampleRef;
