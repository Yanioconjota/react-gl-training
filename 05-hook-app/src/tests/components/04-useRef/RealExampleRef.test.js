import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';

describe('pruebas con RealExampleRef', () => {
  let wrapper = shallow(<RealExampleRef/>);
  test('Debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('debe mostrar el componente <MultipleCustomHooks>', () => {
    const button = wrapper.find("button");
    button.simulate('click');
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
  
  
});
