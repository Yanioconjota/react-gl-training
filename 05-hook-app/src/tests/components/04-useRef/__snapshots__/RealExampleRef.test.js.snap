// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con RealExampleRef Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
